core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -94,7 +107,7 @@ object Queue {
       * @param queue    Queue, expressed as vector for fast cons/uncons from head/tail
       * @param deq      A list of waiting dequeuers, added to when queue is empty
       */
-    case class State(queue: Vector[A], deq: Vector[Async.Ref[F,A]])
+    case class State(queue: Vector[A], deq: Vector[Either[Async.Ref[F,A], Async.Ref[F,Vector[A]]]])
```

Comment:
IIRC, Vector isn't specialized for singletons so you'd have a full 32 element array allocated for each `dequeue1`. Perhaps not that meaningful considering everything else involved in queues but maybe worth considering.